### ISSUE TITLE
[CT-522] [Bug] query-comment should append comment for full-refresh and merge statements

### ISSUE THREAD

**Issue by mkkuba** [dbt-labs/dbt-snowflake#140]
### Describe the bug
query-comment with `append: true` set does NOT append the query comment in Snowflake in the following cases:
- full refresh of a model
- merge statement on incremental updates

### Steps To Reproduce
set query comment config to `append: true`
```
query-comment:
    append: true
```

Run a full refresh of a model
Observe in Snowflake query history that the query-comment is not appended in the query for the full-refresh

Run an incremental update of a model
Observe in Snowflake query history that the query-comment is not appended in the merge statement for the incremental update

### Expected behavior
query-comment should be appended to the statements in the cases outlined above

### Screenshots and log output
If applicable, add screenshots or log output to help explain your problem.

### System information
**The output of `dbt --version`:**
```
installed version: 1.0.1
   latest version: 1.0.4

Your version of dbt is out of date! You can find instructions for upgrading here:
https://docs.getdbt.com/docs/installation

Plugins:
  - snowflake: 1.0.0
```

**The operating system you're using:**
MacOS Monterey 12.3.1

**The output of `python --version`:**
Python 3.9.9

### Additional context
N/A

**Comment by jtcohen6** [comment]
@mkkuba Thanks for opening! You're totally right.

On Snowflake, we need to split up multi-query statements (containing `;`), and pass each one as its own query. The issue here has been that query comments are applied _before_ that split happens: the comment is applied in `execute`, the split occurs in `add_query`. I think the solution here should be relatively straightforward, possibly even related to the fix for #147.

**Comment by mkkuba** [comment]
Thanks @jtcohen6 !
